learn/work/fairness-readmission/index.qmd

```diff
@@ -0,0 +1,517 @@
+* **Construct custom fairness metrics that address the increased variability associated with smaller counts.** We could scale the variability of the estimates for each `race` group before calculating fairness metrics.
+* **Bin the values for infrequent classes.** This would entail collapsing values of `race` other than `"Caucasian"` and `"African American"` into one factor level. This approach is somewhat of a hybrid of the two approaches above; we lose some granularity in information regarding care for `race` groups other than `"Caucasian"` and `"African American"`, but reduce the variability associated with the estimates for those groups in the process.
+
+For this analysis, we'll go with the last option.
+
+::: callout-note
+While we won't construct custom fairness metrics in this example, you can do so using the `new_groupwise_metric()` function in yardstick.
+:::
+
+Recoding that data column:
+
+```{r}
+#| label: "collapse-race"
+readmission_collapsed <-
+  readmission %>%
+  mutate(
+    race = case_when(
+      !race %in% c("Caucasian", "African American") ~ "Other",
+      .default = race
+    ),
+    race = factor(race)
+  )
+
+readmission_collapsed %>%
+  count(race)
+```
+
+Plotting distributions of remaining predictors:
+
+```{r}
+#| label: "plot-age-by-source"
+readmission_collapsed %>%
+  ggplot(aes(x = age)) +
+  geom_bar() +
+  facet_grid(rows = vars(admission_source))
+```
+
+Most patients in this data are in their 60s and 70s. Emergencies account for most admissions in this data, though many others are from referrals or other sources.
+
+```{r}
+#| label: "plot-insurer"
+readmission_collapsed %>%
+  ggplot(aes(x = insurer)) +
+  geom_bar()
+```
+
+While payment information on most patients is missing, most patients in this data are covered under medicare.
+
+::: callout-tip
+The payment method is one way in which societal unfairness may be reflected in the source data besides the variables on protected groups themselves. Medicaid coverage is only available to people below a certain income, and many self-pay patients do not have medical insurance because they cannot afford it. Relatedly, poverty rates differ drastically among racial groups in the U.S.
+:::
+
+```{r}
+#| label: "plot-n-variables"
+#| message: false
+#| warning: false
+readmission_collapsed %>%
+  pivot_longer(starts_with("n_")) %>%
+  ggplot(aes(x = value)) +
+  geom_histogram() +
+  facet_wrap(vars(name), scales = "free_x")
+```
+
+For many patients, this was their first inpatient visit to this hospital system. During their stay, many patients receive 10-20 medications and experience several procedures.
+
+With a sense for the distributions of variables in this dataset, we can move on to splitting data up for modeling.
+
+## Resampling Data
+
+First, splitting data into training and testing:
+
+```{r}
+#| label: "split"
+set.seed(1)
+readmission_splits <- initial_split(readmission_collapsed)
```

Comment:
Want to consider using `strata` since there is some class imbalance?

Reply:
Yes, good thought.

In [80fe7e5](https://github.com/tidymodels/tidymodels.org/pull/52/commits/80fe7e5a195413d647ea411b822a2083b39706a2) I added the argument, but it won't be reflected in the `wflow_set_fit` results until I refit the workflow set once the article is otherwise ready to go. :)